central/declarativeconfig/manager_impl.go

```diff
@@ -8,39 +8,51 @@ import (
 	"time"
 
 	"github.com/gogo/protobuf/proto"
+	"github.com/stackrox/rox/pkg/concurrency"
 	"github.com/stackrox/rox/pkg/declarativeconfig"
 	"github.com/stackrox/rox/pkg/declarativeconfig/transform"
+	"github.com/stackrox/rox/pkg/env"
 	"github.com/stackrox/rox/pkg/k8scfgwatch"
+	"github.com/stackrox/rox/pkg/maputil"
 	"github.com/stackrox/rox/pkg/sync"
 	"github.com/stackrox/rox/pkg/utils"
 )
 
 const (
-	watchInterval        = 5 * time.Second
 	declarativeConfigDir = "/run/stackrox.io/declarative-configuration"
 )
 
-var (
-	// Set Force to true, so we explicitly retry watching the files within the directory and not stop on the first
-	// error occurred.
-	watchOpts = k8scfgwatch.Options{Interval: watchInterval, Force: true}
-)
+type protoMessagesByType = map[reflect.Type][]proto.Message
 
 type managerImpl struct {
-	once                 sync.Once
+	once sync.Once
+
 	universalTransformer transform.Transformer
+
+	transformedMessagesByHandler map[string]protoMessagesByType
+	transformedMessagesMutex     sync.RWMutex
+
+	reconciliationTickerDuration time.Duration
+	watchIntervalDuration        time.Duration
+
+	reconciliationTicker     *time.Ticker
+	reconciliationInProgress concurrency.Flag
+	shortCircuitSignal       concurrency.Signal
 }
 
 // New creates a new instance of Manager.
 // Note that it will not watch the declarative configuration directories when created, only after
-// WatchDeclarativeConfigDir has been called.
+// ReconcileDeclarativeConfigurations has been called.
 func New() Manager {
 	return &managerImpl{
-		universalTransformer: transform.New(),
+		universalTransformer:         transform.New(),
+		transformedMessagesByHandler: map[string]protoMessagesByType{},
+		reconciliationTickerDuration: env.DeclarativeConfigReconcileInterval.DurationSetting(),
```

Comment:
I would expect these parameters to be passed to the `New` method and extracted from `env` in a Singleton